dhall/src/Dhall/Binary.hs

```diff
@@ -377,9 +376,10 @@ encode (DoubleLit n64)
     | otherwise = TDouble n64
   where
     n32      = double2Float n64
-    n16      = floatToWord16 n32
     useFloat = n64 == float2Double n32
-    useHalf  = n64 == (float2Double . wordToFloat16 . fromIntegral) n16
+    -- the other three cases for Half-floats are 0.0 and the infinities
+    useHalf  = any id $ fmap (n64 ==) [0.0, infinity, -infinity]
+    infinity = (read "Infinity") :: Double
```

Comment:
I would personally prefer defining `infinity` via the `Num` interface. But your version via `Read` is fairly understandable too. :smile: 

```suggestion
    infinity = 1/0 :: Double
```